packages/x-components/src/composables/use-state.ts

```diff
@@ -0,0 +1,22 @@
+import { computed } from 'vue';
+import { ExtractState, XModuleName } from '../x-modules/x-modules.types';
+import { useStore } from './use-store';
+
+/**
+ * Generates a computed property which returns the selected state.
+ *
+ * @param module - The {@link XModuleName} of the getter.
+ * @param path - The state path.
+ * @returns The state properties of the module.
+ * @public
+ */
+export function useState<Module extends XModuleName, Path extends keyof ExtractState<Module>>(
```

Comment:
What was the reason to create a new standalone composable instead of extending the `useStore` one? The task gave the two options, so I'm curious about the reasoning.

Reply:
No reason, I had forgotten that option.